agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -220,32 +220,33 @@ public static class DatabaseMaskingConfiguration {
   public static class LoggingInstrumentation {
     public String level = "INFO";
 
-    public Severity getSeverity() {
-      return getSeverity(level);
+    public int getSeverityThreshold() {
+      return getSeverityThreshold(level);
     }
 
-    public static Severity getSeverity(String level) {
+    public static int getSeverityThreshold(String level) {
```

Comment:
_(suggestion)_ This method might be moved in `AgentLogExporter` and called around line 84. So all the log logic would be in the place where we need it. A `boolean exportLog(LogData log, String configuredLogThreshold)` method might then be added and contains all the log level logic for the export. The configuration object would then represent the log level as a string an not an int.

Reply:
this looks a bit more challenging, as we are also using this method for fail-fast validation of the configuration:

https://github.com/microsoft/ApplicationInsights-Java/blob/560c7d58ff29aebcfffa5d88f970edfe0de0a91b/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java#L54-L55